### DISCUSSION TITLE
Can you help me a little with the #2 step of your tutorial.

### DISCUSSION THREAD

**Discussion by proximacentaury** [rzeldent/esp32-smartdisplay#154] [Q&A]
I am strugling to use/repeat your project a weeks.
I have reinstalled everything... (vscode had that "include problem")
But I do not understand well the second step of your instruczion. (It is clear I am not familiar with neither VSCode nor platformio.
I cannot find the way to add that stunton library.

Thank you in advance

**Comment by rzeldent** [comment]
Hi,

The "include problem" is that another repository contains the definitions for the boards. PlatformIO looks for these definitions in the ./boards directory (relative to the project folder). That's why the repository needs to be checked out including the submodules so this directory will be created. These defines are in the https://github.com/rzeldent/platformio-espressif32-sunton project.

There is a tight relation to the smartdisplay library and the definitions. It is also possible to copy these files to the boards directory, but this might break with a new release of the smartdisplay library. So using a submodule, these can be kept in sync by also updating the boards submodule to the matching smartdisplay version...

Hope this explains a bit the reasoning behind these choices...

**Comment by proximacentaury** [comment]
Thank you for your answer... theoretically I understood, bit practiacally I do not know what I need to do (more..)
I try to explai what I did up to now ...

I have deleted almost everything from my computer and reinstalled EVERYTHING. (Like VSCode+Gcc+CMAKE+platformio

I have opened the demo folder.
The remaining error is the :
`UnknownBoard: Unknown board ID 'esp32-2432S024C'`

Can you help me with some advice. 
the board folder has the definitions from platformio-espressiv32-sunton -main

the Platformio / bnoards Explores shows no result on esp32-2432S024C

Thank you in advance

**Comment by proximacentaury** [comment]
> Thank you for your answer... theoretically I understood, bit practiacally I do not know what I need to do (more..) I try to explai what I did up to now ...
> 
> I have deleted almost everything from my computer and reinstalled EVERYTHING. (Like VSCode+Gcc+CMAKE+platformio
> 
> I have opened the demo folder. The remaining error is the : `UnknownBoard: Unknown board ID 'esp32-2432S024C'`
> 
> Can you help me with some advice. the board folder has the definitions from platformio-espressiv32-sunton -main
> 
> the Platformio / bnoards Explores shows no result on esp32-2432S024C
> 
> Thank you in advance

Finally solved!!!
I do not know is this the best way, but it is working,
The solution is even simple...

[https://docs.platformio.org/en/latest/platforms/creating_board.html](url)

I created that boards folder into the  core_dir and copied the esp32-2432S024C.json to that folder.
And it is working.

Best regards.

**Comment by rzeldent** [comment]
It is a solution that works but the file will not be updated if definitions change in the future... 
The preferred solution is to have them in the ./boards directory...
However, nice to hear it works!
